core/src/main/java/de/evoila/cf/config/security/credhub/CredhubConnection.java

```diff
@@ -0,0 +1,62 @@
+package de.evoila.cf.config.security.credhub;
+
+import de.evoila.cf.broker.bean.CredhubBean;
+import de.evoila.cf.config.security.AcceptSelfSignedClientHttpRequestFactory;
+import org.springframework.credhub.core.CredHubClient;
+import org.springframework.credhub.core.CredHubTemplate;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.security.oauth2.client.OAuth2RestTemplate;
+import org.springframework.security.oauth2.client.resource.OAuth2ProtectedResourceDetails;
+import org.springframework.security.oauth2.client.token.AccessTokenProviderChain;
+import org.springframework.security.oauth2.client.token.grant.client.ClientCredentialsAccessTokenProvider;
+import org.springframework.security.oauth2.client.token.grant.client.ClientCredentialsResourceDetails;
+import org.springframework.security.oauth2.common.AuthenticationScheme;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+
+/**
+ * Created by reneschollmeyer, evoila on 15.11.18.
+ */
+@Service
+public class CredhubConnection {
+
+    private static final String GRANT_TYPE="client_credentials";
+
+    private CredhubBean credhubBean;
+
+    public CredhubConnection(CredhubBean credhubBean) {
+        this.credhubBean = credhubBean;
+    }
+
+    public CredHubTemplate createCredhubTemplate() {
+
+        ClientHttpRequestFactory clientHttpRequestFactory = new AcceptSelfSignedClientHttpRequestFactory();
```

Comment:
This is not the way we want it to have. I an configuration like that it automatically accepts SelfSigned for all stages. Please use the same way we do in HaProxyServiceImpl

Reply:
Docs are already in the docs folder on the master branch